client/components/stats-date-control/index.tsx

```diff
@@ -165,58 +148,45 @@ const StatsDateControl = ( {
 
 	return (
 		<div className={ COMPONENT_CLASS_NAME }>
-			{ isCalendarEnabled ? (
-				<DateRange
-					selectedStartDate={ moment( dateRange.chartStart ) }
-					selectedEndDate={ moment( dateRange.chartEnd ) }
-					lastSelectableDate={ moment() }
-					firstSelectableDate={ moment( '2010-01-01' ) }
-					onDateCommit={ ( startDate: Moment, endDate: Moment ) =>
-						startDate &&
-						endDate &&
-						onApplyButtonHandler( startDate.format( 'YYYY-MM-DD' ), endDate.format( 'YYYY-MM-DD' ) )
-					}
-					renderTrigger={ ( {
-						onTriggerClick,
-						buttonRef,
-					}: {
-						onTriggerClick: () => void;
-						buttonRef: RefObject< typeof Button >;
-					} ) => {
-						return (
-							<Button
-								onClick={ () => {
-									const event_from = isOdysseyStats ? 'jetpack_odyssey' : 'calypso';
-									recordTracksEvent( eventNames[ event_from ][ 'trigger_button' ] );
-									onTriggerClick();
-								} }
-								ref={ buttonRef }
-							>
-								{ getButtonLabel() }
-								<Icon className="gridicon" icon={ calendar } />
-							</Button>
-						);
-					} }
-					rootClass="stats-date-control-picker"
-					overlay={ overlay }
-					displayShortcuts
-					useArrowNavigation
-					customTitle="Date Range"
-					focusedMonth={ moment( dateRange.chartEnd ).toDate() }
-					onShortcutClick={ onShortcutClickHandler }
-				/>
-			) : (
-				<DateControlPicker
-					buttonLabel={ getButtonLabel() }
-					dateRange={ dateRange }
-					shortcutList={ shortcutList }
-					selectedShortcut={ getShortcutForRange()?.id }
-					onShortcut={ onShortcutHandler }
-					onApply={ onApplyButtonHandler }
-					onGatedHandler={ onGatedHandler }
```

Comment:
I missed the `onGatedHandler` when migrating the overlay functionality. Likely this is just for dotcom upgrades, but we need to fix it if broken. Not related to the PR tho. Happy for it to merge first.